src/pages/MovieListPage/MovieListPage.js

```diff
@@ -0,0 +1,129 @@
+import { useState, useEffect, useCallback, useMemo } from 'react';
+import { useSearchParams, useNavigate } from 'react-router-dom';
+
+import MovieDetails from '../../components/MovieDetails';
+import GenreSelect from '../../components/GenreSelect';
+import SortControl from '../../components/SortControl';
+import MovieTile from '../../components/MovieTile';
+import useFetch from '../../hooks/useFetch';
+import Header from '../../components/Header';
+import SearchContext from './SearchContext';
+
+const MovieListPage = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const searchQuery = searchParams.get('query') || '';
+  const sortCriterion = searchParams.get('sortBy') || '';
+  const activeGenre = searchParams.get('genre') || '';
+
+  const [movies, setMovies] = useState([]);
+  const [selectedMovie, setSelectedMovie] = useState(null);
+  const sortOrder = sortCriterion === 'title' ? 'asc' : 'desc';
+
+  const url = 'http://localhost:4000/movies';
+  const params = useMemo(() => {
+    return {
+      sortBy: sortCriterion,
+      sortOrder: sortOrder,
+      search: searchQuery,
+      searchBy: 'title',
+      filter: activeGenre,
+    };
+  }, [
+    sortCriterion,
+    sortOrder,
+    searchQuery,
+    activeGenre,
+  ]);
+
+  const navigate = useNavigate();
+  const { loading, error, data: fetchedMovies } = useFetch(url, params);
+
+  useEffect(() => {
+    setMovies(fetchedMovies);
+  }, [fetchedMovies]);
+
+  const handleMovieClick = movie => {
+    navigate(`/movies/${movie.id}`, {
+      state: { searchParams: searchParams.toString() },
+    });
+  };
+
+  const handleBackClick = () => {
+    setSelectedMovie(null);
+  };
+
+  const toggleModal = () => {
+    console.log('This should open the modal to add a movie');
+  };
+
+  const handleSearchQueryChange = value => {
+    setSearchParams(new URLSearchParams({ ...searchParams, query: value }));
+  };
+
+  const handleSortCriterionChange = value => {
+    setSearchParams(prev => {
+      const newParams = new URLSearchParams(prev);
+      newParams.set('sortBy', value);
+      return newParams;
+    });
+  };
+
+  const handleActiveGenreChange = value => {
+    setSearchParams(prev => {
+      const newParams = new URLSearchParams(prev);
+      newParams.set('genre', value);
+      return newParams;
+    });
+  };
+
+  return (
+      <SearchContext.Provider
```

Comment:
1) I'm not sure do you realy need context here. The goal of context is keeping some state but anyway you are keeping it inside the URL

2) It makes sense to keep provider logic separate to component. You can create separate files for provider and context (and maybe some hooks to update context). There is a good article with examples https://kentcdodds.com/blog/how-to-use-react-context-effectively

3) I do not really understand why you keep in context only search query but not rest parameters (genre, sort and so on) since they are equal by the nature. It would be better to keep handle search query and rest parameters together.